### PR TITLE
doc: update cephfs metadata pool recovery procedure

### PR DESCRIPTION
**Pull request by batrick**
This is mostly clarifying a few things are personally experimenting with
this.

Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/cephfs/disaster-recovery-experts.rst [hunk 1]
```diff
@@ -204,63 +204,99 @@ Using an alternate metadata pool for recovery
...
-    cephfs-data-scan init --force-init --filesystem recovery-fs --alternate-pool recovery
-    ceph fs reset recovery-fs --yes-i-really-mean-it
-    cephfs-table-tool recovery-fs:all reset session
-    cephfs-table-tool recovery-fs:all reset snap
-    cephfs-table-tool recovery-fs:all reset inode
+    ceph osd pool create cephfs_recovery_meta
+    ceph fs new cephfs_recovery recovery <data_pool> --allow-dangerous-metadata-overlay
+
+
+The recovery file system starts with a rank that will populate the new metadata
```
Comment (fullerdj) [inline doc/cephfs/disaster-recovery-experts.rst:237]: s/a rank/an MDS rank/
s/populate/initialize/

File: doc/cephfs/disaster-recovery-experts.rst [hunk 2]
```diff
@@ -204,63 +204,99 @@ Using an alternate metadata pool for recovery
...
-    cephfs-data-scan scan_inodes --alternate-pool recovery --filesystem <original file system name> --force-corrupt --force-init <original data pool name>
-    cephfs-data-scan scan_links --filesystem recovery-fs
+    cephfs-data-scan init --force-init --filesystem cephfs_recovery --alternate-pool cephfs_recovery_meta
+    cephfs-data-scan scan_extents --alternate-pool cephfs_recovery_meta --filesystem <fs_name> <data_pool>
+    cephfs-data-scan scan_inodes --alternate-pool cephfs_recovery_meta --filesystem <fs_name> --force-corrupt <data_pool>
+    cephfs-data-scan scan_links --filesystem cephfs_recovery
+
+.. note::
+
+   The scan procedures above go through the entire data pool. This may take a
```
Comment (fullerdj) [inline doc/cephfs/disaster-recovery-experts.rst:265]: s/The/Each/

File: doc/cephfs/disaster-recovery-experts.rst [hunk 3]
```diff
@@ -204,63 +204,99 @@ Using an alternate metadata pool for recovery
...
+    cephfs-journal-tool --rank=<fs_name>:0 event recover_dentries list --alternate-pool cephfs_recovery_meta
+    cephfs-journal-tool --rank cephfs_recovery:0 journal reset --force
 
 After recovery, some recovered directories will have incorrect statistics.
-Ensure the parameters mds_verify_scatter and mds_debug_scatterstat are set
-to false (the default) to prevent the MDS from checking the statistics, then
-run a forward :doc:`scrub </cephfs/scrub>` to repair them. Ensure you have an
-MDS running and issue:
+Ensure the parameters ``mds_verify_scatter`` and ``mds_debug_scatterstat`` are
+set to false (the default) to prevent the MDS from checking the statistics. Next,
```
Comment (fullerdj) [inline doc/cephfs/disaster-recovery-experts.rst:279]: Add the commands here.

File: doc/cephfs/disaster-recovery-experts.rst [hunk 4]
```diff
@@ -204,63 +204,99 @@ Using an alternate metadata pool for recovery
...
+It is recommended to migrate any data from the recovery file system as soon as
+possible. Do not restore the old file system while the recovery file system is
+operational.
+
+.. note::
 
-        ceph daemon mds.a scrub_path / recursive repair
+    If the data pool is also corrupt, some files may not be restored because
+    backtrace information is lost. The data for files will also likely have
+    holes corresponding to the PGs that were lost.
```
Comment (fullerdj) [inline doc/cephfs/disaster-recovery-experts.rst:302]: s/The data.*/If any data objects are missing (due to issues like lost data pool PGs), the recovered files will contain holes in place of the missing data./

**Comment by batrick** [comment]
jenkins render docs

**Comment by batrick** [comment]
https://docs.ceph.com/ceph-prs/35757/cephfs/disaster-recovery-experts/#disaster-recovery-experts

**Comment by batrick** [comment]
jenkins render docs

**Comment by batrick** [comment]
jenkins render docs

**Pull request merged**